api/rest/client.go

```diff
@@ -225,10 +234,10 @@ func (c *Client) sendWithRetries(ctx context.Context, req *http.Request, retryCo
 		c.rateLimiter.Update(ctx, response.StatusCode, response.Header)
 	}
 
-	if c.requestRetrier != nil && retryCount < c.requestRetrier.MaxRetries &&
-		c.requestRetrier.ShouldRetryFunc != nil && c.requestRetrier.ShouldRetryFunc(response) {
-		logger.V(1).Info(fmt.Sprintf("Retrying failed request %q (HTTP %s) after %d ms delay... (try %d/%d)", req.URL, response.Status, 100, retryCount+1, c.requestRetrier.MaxRetries), "statusCode", response.StatusCode, "try", retryCount+1, "maxRetries", c.requestRetrier.MaxRetries)
-		time.Sleep(100 * time.Millisecond)
+	if requestRetrier != nil && retryCount < requestRetrier.MaxRetries &&
+		requestRetrier.ShouldRetryFunc != nil && requestRetrier.ShouldRetryFunc(response) {
```

Comment:
nit:
requestRetrier could offer a method to check for retries e.g. like 
```
func (r *RequestRetrier) ShouldRetry(resp *http.Response) bool {
	if r.ShouldRetryFunc != nil {
		return r.ShouldRetryFunc(resp)
	}
	return false
}
```

```suggestion
		requestRetrier.ShouldRetry(resp) {
```